### DISCUSSION TITLE
Contribute vscode-trace-server repository under eclipse-cdt-cloud/

### DISCUSSION THREAD

**Discussion by marco-miller** [eclipse-cdt-cloud/cdt-cloud#32] [General]
Hi!

So the Trace Compass team at Ericsson would now like to contribute a new extension under here at `eclipse-cdt-cloud`/, named `vscode-trace-server`. This latest extension is a companion to the recently contributed `vscode-trace-extension` [1] down below. It is meant to address the related Issue [2]. This new extension is a VS Code/Codium (OSS) one, alongside [1]; both being typically installed from their respective VSIX package file.

This new extension basically contributes trace server start/stop commands and settings, alongside [1]'s own contributions. There is already a repository for it [3], private with invited collaborators for initial review purposes. Now that its name and main purpose (and implementation) are confirmed, it would be ready for a public move under `eclipse-cdt-cloud`. Goal being, to make this new extension publicly developed, contributed to and available.

Both [1] and [3] are therefore meant as independent extensions. An alternative was to blend [3] into [1], but we preferred to allow [3] to be installable elsewhere or independently from [1]. This allows [3]'s trace server processes to be spawned remotely from [1], collocated with the trace files opened within that server. This is required by the latter, at least for the reference implementation [4]. Trace files may be stored or available elsewhere than where [1] is installed and used. Again, [1] being the actual trace viewer.

As we speak, [3] then has initial commits, along with a few still open Issues and PRs under review. The currently known bug issues were fixed or have a few open PRs to address them. The repository already has a documented README and project standard sibling markdown files based on [1]'s. It has an outstanding PR still under review, about integrating the use of [5] to potentially deploy [3] remotely that way too. The move to `eclipse-cdt-cloud` would carry the clean enough merged commits, along with the still open (and closed) Issues and PRs.

Overall, [3] may then be deployed using combinations of VS Codium, Theia Blueprint (e.g.) and the [5]-related pending PR.

We are about to submit a new Help Desk issue for this move proposal and request. We can invite the necessary extra collaborators for further review access to [3] no problem. So, thanks for any help or feedback herein :)

[1]https://github.com/eclipse-cdt-cloud/vscode-trace-extension
[2]https://github.com/eclipse-cdt-cloud/vscode-trace-extension/issues/15
[3]https://github.com/marco-miller/vscode-trace-server
[4]https://git.eclipse.org/r/plugins/gitiles/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/refs/heads/master/trace-server/
[5]https://github.com/gitpod-io/openvscode-server

**Comment by marco-miller** [comment]
The aforementioned Help Desk issue is now submitted: [6] https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/3229

**Comment by MatthewKhouzam** [comment]
I would like @msujew to weigh in. Typefox may have another solution coming up.

**Reply by MatthewKhouzam** [reply]
Some more info: https://github.com/jonah-iden/theia-native-dependencies

maybe @jonah-iden would be another friend who can comment on this issue.

**Reply by marco-miller** [reply]
This thread only indirectly relates to the hereby repository request, as now confirmed through [7] https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/3229#note_1146455.

Still, these requests for comments remain, should anybody be willing to do so anytime.

**Comment by marco-miller** [comment]
The hereby requested new public repository is now created [8], based on the former private one.

[8] https://github.com/eclipse-cdt-cloud/vscode-trace-server
